lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
after reading https://nodejs.org/api/net.html#net_socket_remoteaddress , I'm -1 on throw here, lose `ip` here mainly cause by client side error(etc: client disconnected), but assert here may record as a server side error(500). 

I'd like default to empty string so the return type of `request.ip` keep the same.